### PR TITLE
Add building Windows Authentication Package to Drone

### PR DESCRIPTION
**Pull request by probakowski**
This change adds building Windows Authentication Package to Drone, both push and tag pipelines.
For tag pipeline it also copies installer so it'll be picked by `relcli` during release artifacts upload.
It also updates `relcli` used by drone to `relcli-master-e148541-20230331T1403513-windows.exe`.
It also bumps `e` to get https://github.com/gravitational/teleport.e/pull/1048 required for build

### CODE REVIEW

**Review by justinas — APPROVED** [review]
Not that familiar with the Windows build stuff, but rlz-related logic looks good to me.

**Review by xacrimon — APPROVED** [review]
Looks correct from a cursory overview but not incredibly familiar.

File: dronegen/windows.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ const (
 	toolchainDir      = `/toolchains`
 	teleportSrc       = `/go/src/github.com/gravitational/teleport`
 
-	relcliURL    = `https://cdn.teleport.dev/relcli-v1.1.76-windows.exe`
-	relcliSha256 = `56dfdd9d1a09aac892fcd48eba035072dc6c151eaa2e1b21cf54786bb3c09520`
+	relcliURL    = `https://cdn.teleport.dev/relcli-master-e148541-20230331T1403513-windows.exe`
```
Comment (zmb3) [inline dronegen/windows.go:26]: Should we be using a versioned build instead of a commit hash?

**Review by zmb3 — APPROVED** [review]

File: dronegen/windows.go (hunk 1 above)
Comment (probakowski) [reply dronegen/windows.go:26]: @justinas this is THE version we use now, right? No v1.x.x?

Comment (probakowski) [reply dronegen/windows.go:26]: I merged it as is, we can revisit it later

Comment (justinas) [reply dronegen/windows.go:26]: Sorry, I missed this conversation. We have stopped tagging vX.Y.Z in Cloud, and as relcli follows Cloud's versioning, we do not have "traditional" versions for relcli either.

**Comment by probakowski** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|branch/v12|[](https://github.com/gravitational/teleport/pull/25311)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**